assets/js/router.js

```diff
@@ -15,25 +15,51 @@ async function fetchPage(url, element) {
                                             </div>`;
                     });
 }
-const links = document.querySelectorAll('.header__nav-item'),
-      linkElems = document.querySelectorAll('.header__nav-link');
+const linkElems = document.querySelectorAll('.header__nav-link');
+
 
 linkElems.forEach((item, i) => {
-    if (item.href.slice(item.href.indexOf('#')) === location.hash){
+    item.addEventListener('click', (event) => {
+        event.preventDefault();
+        const path = event.target.closest('.header__nav-link').getAttribute('href');
+        history.pushState({}, '', path);
+        updatePage(path);
         getLink(i);
-        links[i].click();
-    } else if (location.hash == ''){
-        getLink();
-    }
+    });
 })
 
 function getLink(i = 0){
-    links.forEach(item => item.classList.remove('header__nav-item_active'));
-    links[i].classList.add('header__nav-item_active');
+    linkElems.forEach(item => item.classList.remove('header__nav-link_active'));
+    linkElems[i].classList.add('header__nav-link_active');
```

Comment:
не делаешь проверку на то, что этот элемент будет в списке, хотя бы так сделай
```suggestion
    linkElems[i]?.classList.add('header__nav-link_active');
```

Reply:
поправлено